policy/staged_policy.go

```diff
@@ -88,22 +87,52 @@ func (p StagedPolicies) SamePolicies(other StagedPolicies) bool {
 
 // String is the representation of staged policies.
 func (p StagedPolicies) String() string {
-	var buf bytes.Buffer
-	buf.WriteString(fmt.Sprintf("{cutover:%s,tombstoned:%v,policies:[", time.Unix(0, p.CutoverNanos).String(), p.Tombstoned))
-	for i := range p.policies {
-		buf.WriteString(p.policies[i].String())
-		if i < len(p.policies)-1 {
-			buf.WriteString(",")
-		}
+	b, err := json.Marshal(p)
+	if err != nil {
+		return fmt.Sprintf("[invalid staged policies: %v]", err)
 	}
-	buf.WriteString("]}")
-	return buf.String()
+	return string(b)
 }
 
 func (p StagedPolicies) hasDefaultPolicies() bool {
 	return len(p.policies) == 0
 }
 
+// MarshalJSON returns the JSON encoding of staged policies.
+func (p StagedPolicies) MarshalJSON() ([]byte, error) {
+	return json.Marshal(newStagedPoliciesJSON(p))
+}
+
+// UnmarshalJSON unmarshals JSON-encoded data into staged policies.
+func (p *StagedPolicies) UnmarshalJSON(data []byte) error {
+	var spj stagedPoliciesJSON
+	err := json.Unmarshal(data, &spj)
+	if err != nil {
+		return err
+	}
+	*p = spj.StagedPolicies()
+	return nil
+}
+
+// stagedPoliciesJSON is used for marshaling and unmarshaling staged policies.
+type stagedPoliciesJSON struct {
+	CutoverNanos int64    `json:"cutoverNanos"`
```

Comment:
Unfortunately JSON marshaler doesn't marshal the unexported field `policies` in staged policies.